### ISSUE TITLE
Add Hinge Loss

### ISSUE THREAD

**Issue by tanaymeh** [google-deepmind/optax#403]
Hi, I would like to add Hinge loss to Optax.

I am hoping to contribute more extensively to optax and hence starting with this loss function addition. 

Can I go ahead with this?

Hinge loss: [Wikipedia](https://en.wikipedia.org/wiki/Hinge_loss), [Sklearn](https://scikit-learn.org/stable/modules/generated/sklearn.metrics.hinge_loss.html)

**Comment by mtthss** [comment]
Sounds great! Looking forward to it

**Comment by heytanay** [comment]
@mkunesch Now, that Binary Hinge Loss is merged, can I open a new PR for Multi-class Hinge Loss?

**Comment by mkunesch** [comment]
Totally! Which method exactly would you like to implement? Crammer-Singer similar to sklearn? If possible, it would be great if you could link a reference in the docstring or ideally even write out the equations in the docstring similar to how it's done in Adam:
https://github.com/deepmind/optax/blob/5f0501bac8015a48fc6cf78a6487a714df2b6553/optax/_src/alias.py#L216-L225

Thanks a lot for this and let me know if you have any questions or if you'd like to discuss the implementation!

**Comment by heytanay** [comment]
@mkunesch I think the Crammer-Singer method would be a good option as it will closely resemble the sci-kit-learn's implementation. I will be opening a PR shortly.
